vignettes/v1-2020-2021-mitma-data-codebook.qmd

```diff
@@ -0,0 +1,228 @@
+
+[^1]: For reference: this object also has classes: `tbl_dbi` ,`tbl_sql`, `tbl_lazy` ,and `tbl` .
+
+```{r}
+library(dplyr)
+od_mean_hourly_trips_over_the_4_days <- od_dist |>
+  group_by(time_slot) |>
+  summarise(
+    mean_hourly_trips = mean(n_trips, na.rm = TRUE),
+    .groups = "drop") |> 
+  collect()
+od_mean_hourly_trips_over_the_4_days
+```
+
+```         
+# A tibble: 24 × 2
+   time_slot mean_hourly_trips
+       <int>             <dbl>
+ 1        18              21.4
+ 2        10              19.3
+ 3         2              14.8
+ 4        15              19.8
+ 5        11              19.9
+ 6        16              19.6
+ 7        22              20.9
+ 8         0              18.6
+ 9        13              21.1
+10        19              22.5
+# ℹ 14 more rows
+# ℹ Use `print(n = ...)` to see more rows
+```
+
+In this example above, we calculated mean hourly flows over the 4 days of the requested period. The full data for all 4 days was never loaded into memory all at once. Rather the available memory of the computer was used up to its maximum limit to make that calculation happen, without ever exceeding the available memory limit. This is done transparantly to the user with the help of [`DuckDB`](https://duckdb.org/){target="_blank"} (specifically, with [{duckdb} R package](https://r.duckdb.org/index.html){target="_blank"}).
+
+The same summary operation as provided in the example above can be done with the entire dataset for the full 18 month on a regular laptop with 8-16 GB memory. It will take a bit of time to complete, but it will be done. To speed things up, we will also demonstrate in the end of the document, how the data can be converted to more efficient formats.
+
+***Note**: As long as you use object `od_dist` created with `spod_get_od()` function, it is much quicker to filter the dates by the `year`, `month` and `day` variables, rather than by the `date` variable.* This is because the data for each day is in a separate CSV file located in folders that look like `year=2020/month=2/day=14`. So when filtering by the `date` field, R will have to scan all CSV files comparing the specified date with what is stored inside the file. However, if you query by *`year`, `month` and `day` variables, R only needs to check these against the path to each CSV file, which is much quicker. This caveat is only relevant as long as you use `spod_get_od()` . If you convert (see the [relevant section below](#convert-data)) the downloaded data to a format that it optimized for quick analysis, you can use whichever field you want, it should not affect the performance as much.*
+
+## 2.2. Trips per person data {#tpp-data}
```

Comment:
It took me a while to understand what this subheading meant. I first interpreted it as averages of trips per person. However, the dataset does not show exactly this. The tables in maestra2 show the number of people by a categorization based on number of trips, which can be 0, 1, 2 or +2 by date and zone. Therefore, I would call this section 2.2. People by number of trips or something similar. I mean the relevant or response variable in that dataset for me is the number of people not the number of trips. Does this make sense to you? People by number of trips is still hard to grasp but I think fairer with the content of the content. Do you agree?

Reply:
Yep, I agree, this is confusing right now. I will think of a better formulation based on your suggestions.